src/clue.js

```diff
@@ -2,25 +2,163 @@
+        color: "green"
+    },
+    drOrchid = {
+        
+firstName: "Doctor",
+lastName: "Orchid",
+occupation: "Scientist",
+age: 26,
+description: "PhD in plant toxicology. Adopted daughter of Mr. Boddy",
+image: "http://www.radiotimes.com/uploads/images/Original/111967.jpg",
+color: "white"
+    },
+    profPlum = {
+    
+firstName: "Victor",
+lastName: "Plum",
+occupation: "Designer",
+age: 22,
+description: "Billionaire video game designer",
+image: "https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg",
+color: "purple"
+    },
+    missScarlet = {
+    
+firstName: "Kasandra",
+lastName: "Scarlet",
+occupation: "Actor",
+age: 31,
+description: "She is an A-list movie star with a dark past",
+image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+color: "red"
+  },
 
-const roomsArray = [];
+  mrsPeacock = {
+firstName: "Eleanor",
+lastName: "Peacock",
+occupation: "Socialité",
+age: 36,
+description: "She is from a wealthy family and uses her status and money to earn popularity",
+image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+color: "blue"
+
+  },
+  mrMustard = {
+firstName: "Jack",
+lastName: "Mustard",
+occupation: "Retired Football player",
+age: 62,
+description: "He is a former football player who tries to get by on his former glory",
+image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+color: "yellow"
+  }
+];
+
+
+ // Rooms Array
+
+const roomsArray = [
+     {
+    name: "Dining Room"},{
+    
+    name: "Conservatory"},{
+    name: "Kitchen"},{
+    name: "Study"},{
+    name: "Library"},{
+    name: "Billiard Room"},{
+    name: "Lounge"},{
+    name: "Ballroom"},{
+    name: "Hall"},{
+    name: "Spa"},{
+    name: "Living Room"},{
+    name: "Observatory"},{
+    name: "Theater"},{
+    name: "Guest House"},{
+    name: "Patio"}
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+    {name: "rope",
+    weight: 10},
+    {
+name: "knife",
+weight: 8},
+{
+name: "candlestick",
+weight: 2},
+{
+name: "dumbbell",
+weight: 30},
+{
+name: "poison",
+ weight: 2},
+ {
+name: "axe", 
+weight: 15},
+{
+name: "bat",
+weight: 13},
+{
+name: "trophy",
+ weight: 25},
+ {
+ name: "pistol",  
+weight: 20}];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(array) {
+      const randomIndex = Math.floor(Math.random() * array.length);
+      return array[randomIndex];
+    
+  }
+  
+// ITERATION 3
+
+function pickMystery() {
+    const mystery = {
+    suspect: selectRandom(suspectsArray),
+    weapon: selectRandom(weaponsArray),
+    room: selectRandom(roomsArray)
+    }
+    return mystery
+}
+console.log(pickMystery())
+  
+
+
+function revealMystery(object) {
+    const firstName = object.suspect.Kasandra;
+    const lastName = object.suspect.Scarlet;
+    const weaponUsed = object.weapon.pistol;
+    const place = object.room.GuestHouse;
+  
```

Comment:
The object `suspect` doesn't have a Kasandra or Scarlet properties, but a `firstName` one and a `lastName` too
Object `weapon` does have a property called `name`, and the same goes for object `room`

So, in this case and in order to work your code should look like:

```
function revealMystery(object) {
  const firstName = object.suspect.firstName;
  const lastName = object.suspect.lastName;
  const weaponUsed = object.weapon.name;
  const place = object.room.name;

  return `${firstName} ${lastName} killed Mr. Boddy using the ${weaponUsed} in the ${place}!`;
}
```

The best way though should be like this:

```
function revealMystery(envelope) {
  let { suspect, weapon, room } = envelope;

  return `${suspect.firstName} ${suspect.lastName} killed Mr. Boddy using the ${weapon.name} in the ${room.name}!`;
}
```

It is good practice if you destructure your `envelope` argument.
By destructuring, it clearly expresses which properties you're using within the function.
By extracting the properties into individual variables (suspect, weapon, and room), you can directly reference them in your return statement without needing to use `envelope.suspect`, `envelope.weapon`, and `envelope.room` every time. This simplifies the code and improves readability.